packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -2192,14 +2180,39 @@ class _RenderTheaterMarker extends InheritedWidget {
     ]);
   }
 
+  static _RenderTheaterMarker? maybeOf(
+    BuildContext context, {
+    bool targetRootOverlay = false,
+    bool createDependency = true,
+  }) {
+    if (targetRootOverlay) {
+      final InheritedElement? ancestor = _rootRenderTheaterMarkerOf(
+        LookupBoundary.getElementForInheritedWidgetOfExactType<_RenderTheaterMarker>(context),
+      );
+      assert(ancestor == null || ancestor.widget is _RenderTheaterMarker);
+      if (ancestor == null) {
+        return null;
+      }
+      return context.dependOnInheritedElement(ancestor) as _RenderTheaterMarker;
```

Comment:
it looks like google testing is failing because of this creating a dependency regardless of the `createDependency` flag?